internal/providers/manager/manager.go

```diff
@@ -33,6 +33,12 @@ import (
 
 //go:generate go run go.uber.org/mock/mockgen -package mock_$GOPACKAGE -destination=./mock/$GOFILE -source=./$GOFILE
 
+// NameProviderTuple is a tuple of a provider name and the provider instance
+type NameProviderTuple struct {
+	Name     string
+	Provider v1.Provider
+}
```

Comment:
This is a clear improvement. I was actually wondering about changing the generic provider interface to always have to implement `GetName` and `GetID`. But that's neither here or now.

Reply:
That would be really handy, let's do it in a subsequent PR.